extensions/eclipse-che-theia-plugin-ext/src/node/che-task-service.ts

```diff
@@ -67,6 +71,34 @@ export class CheTaskServiceImpl implements CheTaskService {
             this.clients.splice(idx, 1);
         }
     }
+
+    async fireTaskExited(event: TaskExitedEvent): Promise<void> {
+        let exitedTask: CheTask | undefined;
+        for (const task of this.cheTasks) {
+            try {
+                const runtimeInfo = await task.getRuntimeInfo();
+                if (runtimeInfo.execId === event.execId || runtimeInfo.taskId === event.taskId) {
+                    exitedTask = task;
+
+                    task.fireTaskExited({ taskId: task.id, code: event.code, ctx: runtimeInfo.ctx });
+                    break;
+                }
+            } catch (e) {
+                // allow another hanlers to handle request
+            }
+        }
+
+        if (!exitedTask) {
```

Comment:
If we move following logic to `break` place, maybe code will be more readable. But up to you.

Reply:
Done!